src/sign_dlink_ru.c

```diff
@@ -183,15 +187,27 @@ FILE* make_out_file(char* filename) {
  * Sign the firmware file after all of our checks have completed
  */
 void sign_firmware(char* filename, char** salts, int num_salts) {
-	int i;
+	int i = 0;
 	salt_t salt;
 	FILE* f = make_out_file(filename);
+	
+	char* version_suffix = "c0ffeef0rge";
+			
 
+	if (strcmp(salts[0],"-v") == 0) {
+		init_salt(&salt, salts[1]);
+		version_suffix = convert_hex_to_bin(salts[1]);
```

Comment:
The code expects that version_suffix is a null terminated string. If the user specified hex doesn't end in null termination, `add_version_suffix` will run past the end.

Reply:
another C thing i'm not up with
it's only in hex due to the check for hex is already in the progam and seemd very flexible